### ISSUE TITLE
Default value callback ignored for Code json field.

### ISSUE THREAD

**Issue by flexchar** [laravel/nova-issues#2977]
- Nova Version: 3.9.0 (worked in 3.8.4)
- PHP Version: 7.4

### Description:

It's quite simple.
I have setup where I use Code field to display JSON data in an action.

```php
Code::make(
                'Template',
                fn() => collect([...])->toJson(),
            )

```
`resolveUsing` is also not working anymore...

I couldn't pin point what has changed between these two releases but something did...

But if I do this - it works:
```php
$json = collect([...])->toJson();
Code::make(
                'Template',
                $json,
            )
```

**Comment by crynobone** [comment]
Quick question, based on the issue it seem that the issue is mainly on using Field on Action. So wouldn't the following be used instead?

```php
Code::make('Template')
    ->default(n() => collect([...])->toJson())
    ->json(),
```

Will attempt to create a reproducing code repo later.

**Comment by crynobone** [comment]
![image](https://user-images.githubusercontent.com/172966/96668773-75f1d980-138e-11eb-8cda-1f82ead1ec91.png)

```php
    /**
     * Get the fields available on the action.
     *
     * @return array
     */
    public function fields()
    {
        return [
            Code::make('Snippet (2nd parameter)', function () {
                return collect([1, 2, 3])->toJson();
            }),

            Code::make('Snippet (resolveUsing)')->resolveUsing(function () {
                return collect([1, 2, 3])->toJson();
            }),

            Code::make('Snippet (displayUsing)')->displayUsing(function () {
                return collect([1, 2, 3])->toJson();
            }),

            Code::make('Snippet (default)')->default(function () {
                return collect([1, 2, 3])->toJson();
            }),
        ];
    }
```

**Comment by crynobone** [comment]
![image](https://user-images.githubusercontent.com/172966/96669284-80f93980-138f-11eb-9eee-f3f3f37e84c0.png)

```php
    /**
     * Get the fields available on the action.
     *
     * @return array
     */
    public function fields()
    {
        return [
            Text::make('Snippet (2nd parameter)', function () {
                return collect([1, 2, 3])->toJson();
            }),

            Text::make('Snippet (resolveUsing)')->resolveUsing(function () {
                return collect([1, 2, 3])->toJson();
            }),

            Text::make('Snippet (displayUsing)')->displayUsing(function () {
                return collect([1, 2, 3])->toJson();
            }),

            Text::make('Snippet (default)')->default(function () {
                return collect([1, 2, 3])->toJson();
            }),
        ];
    }
```

**Comment by crynobone** [comment]
Based on my tests the behaviour is consistent between fields, hence I would argue the previous working code behaviour has changed. 

Please update the code to use `default()` when defining Field for Action.

Code example: https://github.com/nova-issues/issue-2977/commit/3bef8f3fa8b137e13816475a0811e80ec089a5c4

**Comment by flexchar** [comment]
@crynobone  true. I was mainly using it in action and with `resolveValue`. I never heard of `default()` perhaps it got after I implemented but it used to work just fine before the mentioned release. Anyway, thank you for clarifying! :)

**Comment by crynobone** [comment]
> but it used to work just fine before the mentioned release.

Yup, the previous method no longer work. You can refer to laravel/nova#941 which change the behaviour to fix #2796

I would argue using `default()` is better here because Action Fields is meant only for generating form for Action, so `resolveUsing()` or `displayUsing()` (which mainly to customize attribute value from Eloquent Model) seem rather hackish. 

I have added a PR to the Nova Documentation regarding this. laravel/nova-docs#260

**Comment by flexchar** [comment]
Super man. High five for the documentation!
